src/dataflow/src/logging/timely.rs

```diff
@@ -477,6 +447,18 @@ pub fn construct<A: Allocate>(
     traces
 }
 
+fn create_address_row(id: i64, worker: i64, address: &[usize]) -> Row {
+    let mut address_row = Row::default();
+    address_row.push(Datum::Int64(id));
+    address_row.push(Datum::Int64(worker));
+
+    // TODO(aljoscha): figure out how to use Row::pack_slice when
+    // there's a list in there
```

Comment:
Probably impossible to make `pack_slice` work here. I'd recommend using the `datum_size` function to precompute the size and then use `Row::with_capacity`. You'll probably need to add a `datum_list_size` function that just computes (1 + sizeof(usize) + datum_size(iter)) because you won't be able to construct a `Datum::List` ahead of time.

Reply:
Thanks, I pushed a fixup. I use `clone()` on an iterator in there but in that case it's fine, because it only clones the iterator, not the underlying data, right?